Node/src/TeamsChatConnector.ts

```diff
@@ -150,22 +163,41 @@ export class TeamsChatConnector extends builder.ChatConnector {
     for (var event of events) {
       let invoke = <IInvokeEvent>event;
       if (invoke.type == 'invoke') {
+        let query = <ComposeExtensionQuery>(<any>event).value;
+        let handler: ComposeExtensionHandlerType;
         switch (invoke.name) {
-          case 'composeExtension/query':
-            this.dispatchQuery(invoke, callback);
+          case TeamsChatConnector.queryInvokeName:
+            handler = this.dispatchQuery;
+            break;
+          case TeamsChatConnector.querySettingUrlInvokeName:
+            handler = this.querySettingUrlHandler;
+            break;
+          case TeamsChatConnector.settingInvokeName:
+          {
+            handler = this.settingHandler;
             break;
+          }
           default:
             realEvents.push(event);
             break;
         }
+        if (handler) {
+          try {
+            handler(invoke, query, callback);
+          }
+          catch (e) {
+            console.log(e);
+            callback(e, null, 500);
+          }
+        }
       }
       else {
         realEvents.push(event);
       }
     }
     if (realEvents.length > 0) {
       // Filter out @mention
-      var out: builder.IEvent[] = []; 
+      var out: builder.IEvent[] = [];
```

Comment:
I think you should start a separate PR to clean up the var vs let in this file. My editor was just trimming white spaces in this file.